src/modules/auth/strategy/api-key.strategy.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { AUTH } from '@ukef/constants';
+import { HeaderAPIKeyStrategy } from 'passport-headerapikey';
+
+import { AuthService } from '../auth.service';
+
+@Injectable()
+export class ApiKeyStrategy extends PassportStrategy(HeaderAPIKeyStrategy, AUTH.STRATEGY) {
+  constructor(private readonly authService: AuthService, private readonly configService: ConfigService) {
+    const headerKeyApiKey: string = configService.get<string>('app.apiKeyStrategy');
+    super({ header: headerKeyApiKey, prefix: '' }, true, (apiKey: string, done: (arg0: UnauthorizedException, arg1: boolean) => void) => {
+      const hasValidKey = this.authService.validateApiKey(apiKey);
+      if (hasValidKey) {
+        return done(null, true);
+      }
+      done(new UnauthorizedException(), null);
+    });
+  }
+}
```

Comment:
We discussed this on [the MDM API already](https://github.com/UK-Export-Finance/mdm-api/pull/44#discussion_r1126387816), but it would be good to unit test this

Reply:
that's right. I've tried to look at this today but I keep getting distracted. Can I get your eyes on it to add the test?

Reply:
We have added the tests now